nerfstudio/exporter/exporter_utils.py

```diff
@@ -331,7 +343,14 @@ def collect_camera_poses(pipeline: VanillaPipeline) -> Tuple[List[Dict[str, Any]
     eval_dataset = pipeline.datamanager.eval_dataset
     assert isinstance(eval_dataset, InputDataset)
 
-    train_frames = collect_camera_poses_for_dataset(train_dataset)
+    from nerfstudio.models.splatfacto import SplatfactoModel
+
+    camera_optimizer = None
+    if isinstance(pipeline.model, SplatfactoModel):
```

Comment:
This is a custom code for SplatfactoModel. But I also have Nerfacto model which have camera optimizer enabled by default. Can we support both?

Reply:
Hmm. Good point. In theory, yes, but if we want a generic solution instead of separate ifs for Spaltfacto and Nerfacto, we would also need to fix this: https://github.com/nerfstudio-project/nerfstudio/issues/3073

Reply:
Maybe we should address #3073 first, and use the same API call for apply_to_camera()? The additional change seems very manageable in my opinion. 

> keep the current behavior for NeRFs, but match the splatfacto behavior for consistency

Reply:
Ok. I'll get back to this a bit later

Reply:
@jb-ye Done: I tested that I now can:
 1. optimize poses using either Splatfacto or Nerfacto (with `--eval-mode all` and `--pipeline.model.camera-optimizer.mode=SO3xR3`)
 2. `ns-export cameras`
 3. modify the input data by replacing the original noisy poses with the optimized poses
 4. train Splatfacto on the modified input data without pose optimization, and get good results

My scripts for step 3 are a bit messy (based on [this](https://github.com/SpectacularAI/3dgs-deblur/blob/main/combine.py), mostly), but at least in principle, it now works.